app/src/main/java/com/appunite/loudius/ui/pullrequests/PullRequestsScreen.kt

```diff
@@ -57,50 +57,67 @@ fun PullRequestsScreen(
     navigateToReviewers: NavigateToReviewers,
 ) {
     val state = viewModel.state
-    LaunchedEffect(state.navigateToReviewers) {
-        state.navigateToReviewers?.let {
-            navigateToReviewers(it.owner, it.repo, it.pullRequestNumber, it.submissionTime)
-            viewModel.onAction(PulLRequestsAction.OnNavigateToReviewers)
-        }
-    }
     PullRequestsScreenStateless(
-        pullRequests = state.pullRequests,
+        state = state,
         onAction = viewModel::onAction,
-        isLoading = state.isLoading,
-        isError = state.isError,
+        navigateToReviewers = navigateToReviewers,
     )
 }
 
 @Composable
 private fun PullRequestsScreenStateless(
-    pullRequests: List<PullRequest>,
+    state: PullRequestState,
     onAction: (PulLRequestsAction) -> Unit,
-    isLoading: Boolean,
-    isError: Boolean,
+    navigateToReviewers: NavigateToReviewers,
 ) {
     Scaffold(
         topBar = {
             LoudiusTopAppBar(title = stringResource(R.string.app_name))
         },
         content = { padding ->
-            when {
-                isError -> LoudiusFullScreenError(
+            when (state) {
+                is PullRequestState.Error -> LoudiusFullScreenError(
                     modifier = Modifier.padding(padding),
                     onButtonClick = { onAction(PulLRequestsAction.RetryClick) },
                 )
 
-                isLoading -> LoudiusLoadingIndicator(Modifier.padding(padding))
-                pullRequests.isEmpty() -> EmptyListPlaceholder(padding)
-                else -> PullRequestsList(
-                    pullRequests = pullRequests,
-                    modifier = Modifier.padding(padding),
-                    onItemClick = onAction,
-                )
+                is PullRequestState.Loading -> LoudiusLoadingIndicator(Modifier.padding(padding))
+                is PullRequestState.Loaded -> {
+                    PullRequestContent(state, padding, onAction)
+                    LaunchedEffect(state.navigateToReviewers) {
+                        state.navigateToReviewers?.let {
+                            navigateToReviewers(
+                                it.owner,
+                                it.repo,
+                                it.pullRequestNumber,
+                                it.submissionTime
+                            )
+                            onAction(PulLRequestsAction.OnNavigateToReviewers)
+                        }
```

Comment:
**Info:** In that case it would be probably better to move navigation events to a separate flow.